### PR TITLE
added the error-code info command

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
ever encountered an ignored error in the pack-ignore file and didn't know what it means?
use the `demisto-sdk error-code` command to quickly find relevant information regarding this error.
Note: this command is hidden.

## Screenshots
image

### CODE REVIEW

File: demisto_sdk/__main__.py [hunk 1]
```diff
@@ -1602,6 +1603,28 @@ def convert(config, **kwargs):
     sys.exit(0)
 
 
+@main.command(
+    name='error-code',
+    help='',
```
Comment (roysagi) [inline demisto_sdk/__main__.py:1608]: please do add a help script.

File: demisto_sdk/commands/error_code_info/error_code_info.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+TEMPLATE = '''
+Error Code: {code}
+Function: {func}
+Message:
+{msg}
+'''
+
+
+def print_error_information(func_name, error_data, func, sig):
+    print(f'{Fore.GREEN}## Error Code Info ##{Fore.WHITE}')
```
Comment (roysagi) [inline demisto_sdk/commands/error_code_info/error_code_info.py:15]: why not click.secho?

File: demisto_sdk/commands/error_code_info/error_code_info.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        msg=func(**{a: f'<{a}>' for a in list(sig.parameters)}),
+    ))
+
+
+def find_error(error_code):
+    for func_name, error_data in errors.ERROR_CODE.items():
+        if error_data['code'] == error_code:
+            return func_name, error_data
+
+    return '', {}
```
Comment (roysagi) [inline demisto_sdk/commands/error_code_info/error_code_info.py:28]: you can use `get_error_object` found in errors.py

File: demisto_sdk/commands/error_code_info/error_code_info.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+#     help='The error code to search for.',
+# )
+# def error_code_info(config, **kwargs):
+#     check_configuration_file('error-code-info', kwargs)
+#     sys.path.append(config.configuration.env_dir)
+#
+#     generate_error_code_information(kwargs.get('input'))
+#
+#     sys.exit(0)
+#
```
Comment (roysagi) [inline demisto_sdk/commands/error_code_info/error_code_info.py:62]: remove please

**Review by roysagi — CHANGES_REQUESTED** [review]
Please add a quick test for this.

File: demisto_sdk/commands/error_code_info/error_code_info.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+{msg}
+'''
+
+
+def print_error_information(func_name, error_data, func, sig):
+    print(f'{Fore.GREEN}## Error Code Info ##{Fore.WHITE}')
+    print(TEMPLATE.format(
+        code=error_data['code'],
+        func=f'{func_name}{sig}',
+        msg=func(**{a: f'<{a}>' for a in list(sig.parameters)}),
```
Comment (yaakovi) [inline demisto_sdk/commands/error_code_info/error_code_info.py:19]:
```suggestion
        msg=func(**{a: f'<{a}>' for a in list(sig.parameters)})[0],
```

File: demisto_sdk/commands/error_code_info/error_code_info.py (hunk 3 above)
Comment (yaakovi) [reply demisto_sdk/commands/error_code_info/error_code_info.py:28]: thought about it but it doesn't return the function name.
I didn't want to change the signature of it.

File: demisto_sdk/commands/error_code_info/error_code_info.py (hunk 2 above)
Comment (yaakovi) [reply demisto_sdk/commands/error_code_info/error_code_info.py:15]: changed

**Review by roysagi — APPROVED** [review]

**Pull request merged**